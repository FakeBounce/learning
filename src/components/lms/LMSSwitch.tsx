import { styled } from '@mui/system';
import { Switch } from '@mui/material';
import { ChangeEvent } from 'react';

const AntSwitch = styled(Switch)(({ theme }) => ({
  width: 35,
  height: 19,
  padding: 0,
  display: 'flex',
  '&:active': {
    '& .MuiSwitch-thumb': {
      width: 18
    },
    '& .MuiSwitch-switchBase.Mui-checked': {
      transform: 'translateX(9px)'
    }
  },
  '& .MuiSwitch-switchBase': {
    padding: 2,
    '&.Mui-checked': {
      transform: 'translateX(16px)',
      color: '#fff',
      '& + .MuiSwitch-track': {
        opacity: 1,
        backgroundColor: theme.palette.primary.main
      }
    }
  },
  '& .MuiSwitch-thumb': {
    boxShadow: '0 2px 4px 0 rgb(0 35 11 / 20%)',
    width: 15,
    height: 15,
    borderRadius: 8
  },
  '& .MuiSwitch-track': {
    borderRadius: 16,
    opacity: 1,
    backgroundColor: 'rgba(0,0,0,.25)',
    boxSizing: 'border-box'
  }
}));

interface LMSSwitchProps {
  handleChange: (_: ChangeEvent<HTMLInputElement>) => void;
  checked: boolean;
}

export default function LMSSwitch({ handleChange, checked }: LMSSwitchProps) {
  return <AntSwitch onChange={handleChange} checked={checked} />;
}
