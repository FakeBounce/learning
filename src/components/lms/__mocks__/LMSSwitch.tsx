import { ChangeEvent } from 'react';
import { SwitchProps } from '@mui/material';

interface LMSSwitchProps extends SwitchProps {
  handleChange: (_: ChangeEvent<HTMLInputElement>) => void;
  checked: boolean;
}

export default function LMSSwitch({
  handleChange,
  checked,
  name,
  disabled = false
}: LMSSwitchProps) {
  return (
    <input
      type="checkbox"
      onChange={handleChange}
      checked={checked}
      name={name}
      id={name}
      disabled={disabled}
    />
  );
}
