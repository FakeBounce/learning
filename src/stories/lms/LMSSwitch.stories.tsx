import type { Meta, StoryFn } from '@storybook/react';
import LMSSwitch from '@src/components/lms/LMSSwitch';
import { ChangeEvent, useState } from 'react';

const meta: Meta = {
  title: 'Example/LMSSwitch',
  component: LMSSwitch,
  tags: ['autodocs'],
  parameters: {
    layout: 'centered'
  }
};

export default meta;

const Template: StoryFn<typeof LMSSwitch> = (args: any) => {
  const [checked, setChecked] = useState(args.checked || false);

  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
    setChecked(event.target.checked);
  };

  return <LMSSwitch {...args} checked={checked} handleChange={handleChange} />;
};

export const Default = Template.bind({});
Default.args = {};

export const Checked = Template.bind({});
Checked.args = {
  checked: true
};
