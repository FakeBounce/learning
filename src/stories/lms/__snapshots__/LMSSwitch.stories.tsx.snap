// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example/LMSSwitch Checked smoke-test 1`] = `
<span class="MuiSwitch-root MuiSwitch-sizeMedium css-tvdqvb-MuiSwitch-root">
  <span class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-checked css-1brz8fh-MuiButtonBase-root-MuiSwitch-switchBase">
    <input class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
           type="checkbox"
           checked
    >
    <span class="MuiSwitch-thumb css-tqmoez-MuiSwitch-thumb">
    </span>
    <span class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root">
    </span>
  </span>
  <span class="MuiSwitch-track css-1yjjitx-MuiSwitch-track">
  </span>
</span>
`;

exports[`Example/LMSSwitch Default smoke-test 1`] = `
<span class="MuiSwitch-root MuiSwitch-sizeMedium css-tvdqvb-MuiSwitch-root">
  <span class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-1brz8fh-MuiButtonBase-root-MuiSwitch-switchBase">
    <input class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
           type="checkbox"
    >
    <span class="MuiSwitch-thumb css-tqmoez-MuiSwitch-thumb">
    </span>
    <span class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root">
    </span>
  </span>
  <span class="MuiSwitch-track css-1yjjitx-MuiSwitch-track">
  </span>
</span>
`;
